pkg/virtual/apiexport/builder/build.go

```diff
@@ -100,19 +102,41 @@ func BuildVirtualWorkspace(
 					return nil, fmt.Errorf("error getting valid cluster from context: %w", err)
 				}
 
+				user, found := genericapirequest.UserFrom(ctx)
+				if !found {
+					return nil, fmt.Errorf("error getting user from context")
+				}
+
 				// Wildcard requests cannot be impersonated against a concrete cluster.
 				if cluster.Wildcard {
 					return dynamicClient, nil
 				}
 
-				impersonationConfig := rest.CopyConfig(cfg)
-				impersonationConfig.Impersonate = rest.ImpersonationConfig{
-					UserName: "system:serviceaccount:default:rest",
-					Groups:   []string{bootstrap.SystemKcpAdminGroup},
+				// Add a warrant of a fake local service account giving full access
+				warrant := validation.Warrant{
+					User:   "system:serviceaccount:default:rest",
```

Comment:
Just of interest, what is somebody creates an account like this?

Reply:
totally possible. Same as today: you can create any kind of client cert faking a service account or oidc user or whatever you like. In other words, control over an authn provider is always critical.

Reply:
But don't think you can impersonate this in any way.

Reply:
Impersonation is a good point. Please double check that this is properly e2e tested. Escalating permissions due to impersonation in a workspace must not be possible.

Reply:
deal. will try to spin some tests scenarios for this manually too.